packages/compiler/test/checker/templates.test.ts

```diff
@@ -60,7 +60,7 @@ describe("compiler: templates", () => {
     const diagnostics = await testHost.diagnose("main.tsp");
     strictEqual(diagnostics.length, 1);
     strictEqual(diagnostics[0].code, "invalid-template-args");
-    strictEqual(diagnostics[0].message, "Too few template arguments provided.");
+    strictEqual(diagnostics[0].message, "Template argument 'T' is required and not specified.");
```

Comment:
Shouldn't that message stay the same? If we are missing positional parameters? Feels like otherwise this error leads you to believe we are only missing template argument `T` while we might be missing more.

Reply:
I struggled with this a little bit. You'll at least get one error _per uninstantiated parameter name_. When I thought about it, it just didn't seem like "too few" was really the right error anymore, for example:

Given `alias Foo<T, U = string>`, `Foo<U = "bar">`... well now I don't exactly have "too few" arguments. I have enough arguments; they're just the wrong ones and T is unbound.

I could also specify a list of args with bogus names in a template with different names, and then the problem wouldn't be that I have too few of them either:

Given `alias Foo<T, U, V>`, `Foo<X = string, Y = numeric, Z = boolean>`... I have enough args, they're just all wrong, so "too few" is again not quite the right error anymore.

I eventually landed on just raising one `Missing required template argument` error per missing arg, but I'm open to alternatives, just don't think "too few" is the one anymore.

Reply:
hhm I guess that's what you get in csharp when missing an argument on a function 
<img width="818" alt="image" src="https://github.com/microsoft/typespec/assets/1031227/e99ee853-25af-4ca5-ab12-ddb7c195c2a5">